go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/GoogleCloudPlatform/k8s-cloud-provider
 
-go 1.21
```

Comment:
do this change in separate PR

Reply:
why? this is the first PR that requires a newer version and it doesn't mean anything besides "I need stuff from at least this version". PR for one-line change seems like an overkill

Reply:
If I'd do so, tests would fail.

Reply:
tests are still failing, what needs 1.22. maps.Equal is in 1.21
https://screenshot.googleplex.com/5VEAYaUxvSdLLKQ.png

Reply:
I will try test the old version with the new image

Reply:
I think you meant to show `1.21` not `1.22.1`. However, you are fully correct. Go version bump is not needed for `maps.Equal` (or anything else in this PR).

Reply:
Done.